core/modules/util/threadSafe.h

```diff
@@ -73,21 +73,23 @@ class Flag {
 
     /** Sets flag value to 'val' and returns the old value of flag.
      */
-    virtual T set(T val) {
+    virtual T exchange(T val) {
```

Comment:
Well, my view of that is that:
- it should not be used as atomic, because it's very inefficient
- in many other cases when we need to have locks for extended periods we are already using standard mutex classes so there is little benefit for having this separate class
- many people (me including) think that recursive mutex is an indication of a bad/inefficient design